### PR TITLE
Sparkt zookeeper integration #144

### PR DESCRIPTION
**Pull request by anistal**
*Created Factory using Apache Curator Framework.
*Spec to test  the factory.

### CODE REVIEW

File: driver/src/main/scala/com/stratio/sparkta/driver/factory/CuratorFactoryHolder.scala [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+ * @author anistal
+ */
+object CuratorFactoryHolder extends SLF4JLogging {
+
+  private var curatorFramework: Option[CuratorFramework] = None
+
+  /**
+   * Default ZK connection parameters when no config was specified.
+   */
+  val DefaultConnectionString   =   "localhost:2181"
```
Comment (sgomezg) [inline driver/src/main/scala/com/stratio/sparkta/driver/factory/CuratorFactoryHolder.scala]: Please, don't use default config values in code!

File: driver/src/main/scala/com/stratio/sparkta/driver/Sparkta.scala [hunk 2]
```diff
@@ -63,5 +66,8 @@ object Sparkta extends App with SLF4JLogging {
 
   IO(Http) ! Http.Bind(controller, interface = apiConfig.getString("host"), port = apiConfig.getInt("port"))
 
+//  log.info("Starting ZK ...")
```
Comment (sgomezg) [inline driver/src/main/scala/com/stratio/sparkta/driver/Sparkta.scala]: Remove commented code, please.

File: driver/src/test/scala/com/stratio/sparkta/driver/test/factory/CuratorFactoryHolderSpec.scala [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+    Given(s"ZK configuration: $CuratorFactoryHolderSpec.configString")
+    When("an instance is created with the CuratorFactoryHolder")
+    val instance = CuratorFactoryHolder.getInstance(CuratorFactoryHolderSpec.basicConfig)
+    And("other instance is created with the CuratorFactoryHolder")
+    val secondInstance = CuratorFactoryHolder.getInstance(CuratorFactoryHolderSpec.basicConfig)
+    Then("the factory return the same instance for both cases")
+    instance should be theSameInstanceAs secondInstance
+  }
+
+  "The curator ZK client" must "not reuse  the same connecton when resetInstance is invoked" in {
```
Comment (sgomezg) [inline driver/src/test/scala/com/stratio/sparkta/driver/test/factory/CuratorFactoryHolderSpec.scala]: You have to specify the name only the first time. The next test cases should/can start with "it"

File: driver/src/main/scala/com/stratio/sparkta/driver/factory/CuratorFactoryHolder.scala (hunk 1 above)
Comment (anistal) [inline driver/src/main/scala/com/stratio/sparkta/driver/factory/CuratorFactoryHolder.scala]: Which is the alternative to put them?

Comment (sgomezg) [inline driver/src/main/scala/com/stratio/sparkta/driver/factory/CuratorFactoryHolder.scala]: I think it should be in config.properties.template or something like this. What do you think?

File: driver/src/test/scala/com/stratio/sparkta/driver/test/factory/CuratorFactoryHolderSpec.scala [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+import org.junit.runner.RunWith
+import org.scalatest.junit.JUnitRunner
+import org.scalatest.{FlatSpec, _}
+
+/**
+ * This test specifies the behaviour of CuratorFactoryHolder that encapsulates the real curator's factory.
+ * @author anistal
+ */
+@RunWith(classOf[JUnitRunner])
+class CuratorFactoryHolderSpec extends FlatSpec with Matchers with BeforeAndAfter with GivenWhenThen with SLF4JLogging {
```
Comment (danielcsant) [inline driver/src/test/scala/com/stratio/sparkta/driver/test/factory/CuratorFactoryHolderSpec.scala]: Should't be this class an integration test? In that case it should be renamed to CuratorFactoryHolderIT

File: driver/src/main/scala/com/stratio/sparkta/driver/factory/CuratorFactoryHolder.scala (hunk 1 above)
Comment (anistal) [inline driver/src/main/scala/com/stratio/sparkta/driver/factory/CuratorFactoryHolder.scala]: IMHO I think that a default configuration should not change over time and it is something static. Furthermore, if the project has a lot of default configuration files could be confuse to the user.

File: driver/src/main/scala/com/stratio/sparkta/driver/Sparkta.scala (hunk 2 above)
Comment (anistal) [inline driver/src/main/scala/com/stratio/sparkta/driver/Sparkta.scala]: done.

File: driver/src/test/scala/com/stratio/sparkta/driver/test/factory/CuratorFactoryHolderSpec.scala (hunk 4 above)
Comment (anistal) [inline driver/src/test/scala/com/stratio/sparkta/driver/test/factory/CuratorFactoryHolderSpec.scala]: Done.

**Comment by ajnavarro** [comment]
We're using the [typesafe config library](https://github.com/typesafehub/config). IMHO you must add the default configuration into the reference.conf in https://github.com/Stratio/sparkta/blob/master/driver/src/main/resources/reference.conf and if you want also here: https://github.com/Stratio/sparkta/blob/master/dist/src/main/resources/application.conf. more info [here](https://github.com/typesafehub/config#standard-behavior)

**Comment by anistal** [comment]
Default values in code were deleted; these values must be declared at least in reference.conf as it is described in Typesafe's documentation. 

Thanks @sgomezg and @ajnavarro for your suggestions :)

**Comment by sgomezg** [comment]
LGTM :)

**Pull request merged**